components/service/glean/src/main/java/mozilla/components/service/glean/utils/FreshBootDetector.kt

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.utils
+
+import android.content.Context
+import android.os.Build
+import android.provider.Settings
+import android.support.annotation.VisibleForTesting
+import java.io.File
+import java.io.IOException
+import java.util.UUID
+
+internal object FreshBootDetector {
+    // cache the value of the first check so we can safely run it multiple times
+    @VisibleForTesting
+    var cacheIsFreshBoot: Boolean? = null
+
+    /**
+     * Gets the boot id, which is a value that changes every time the
+     * device is rebooted.
+     *
+     * For Android Oreo (SDK 24) and later, we can use the [BOOT_COUNT] global
+     * setting.
+     *
+     * For earlier versions of Android, we can use the Linux kernel's
+     * `/proc/sys/kernel/random/boot_id`.
+     *
+     * Adapted from https://github.com/instacart/truetime-android/pull/93
+     *
+     * @param context An application context
+     */
+    internal fun getBootId(context: Context): String {
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.O) {
+            try {
+                return Settings.Global.getInt(
+                    context.getContentResolver(),
+                    Settings.Global.BOOT_COUNT
+                ).toString()
+            } catch (e: Settings.SettingNotFoundException) {
+                // pass through to old method
```

Comment:
In the linked discussion it sounds like accessing `boot_id` doesn't work anymore on Android M+ ... does it make sense to pass through?

Reply:
Yeah, perhaps not.  I tried to find a reference to that `/proc` file being removed or permission-locked, but couldn't find any.  I found a bunch of references to [other parts of proc being locked down](https://issuetracker.google.com/issues/37140047), though.